src/test/playwright/e2e/exercise/programming/ProgrammingExerciseParticipation.spec.ts

```diff
@@ -72,30 +74,182 @@ test.describe('Programming exercise participation', () => {
+                exercise,
+                firstSubmission.student,
+                firstSubmission.submission,
+                firstSubmission.commitMessage,
+            );
+
+            for (let i = 1; i < submissions.length; i++) {
+                const { student, submission, commitMessage } = submissions[i];
+                await login(student, '/');
+                await page.waitForURL(/\/courses/);
+                await courseList.openCourse(course.id!);
+                await courseOverview.openExercise(exercise.title!);
+                submission.deleteFiles = [];
+                await makeGitExerciseSubmission(page, programmingExerciseOverview, course, exercise, student, submission, commitMessage);
+            }
+
+            await login(studentFour, '/');
+            await page.waitForURL(/\/courses/);
+            await courseList.openCourse(course.id!);
+            await courseOverview.openExercise(exercise.title!);
+            await expect(programmingExerciseOverview.getCloneRepositoryButton()).not.toBeVisible();
+        });
+
+        test('Students without a team can not participate in the team exercise', async ({ login, page, courseList, courseOverview, programmingExerciseOverview }) => {
+            await login(studentFour, '/');
+            await page.waitForURL(/\/courses/);
+            await courseList.openCourse(course.id!);
+            await courseOverview.openExercise(exercise.title!);
+            await expect(programmingExerciseOverview.getExerciseDetails().getByText('No team yet')).toBeVisible();
+            await expect(courseOverview.getStartExerciseButton(exercise.id!)).not.toBeVisible();
+            await expect(programmingExerciseOverview.getCloneRepositoryButton()).not.toBeVisible();
+        });
+
+        test('Students of other teams have their own submission', async ({
+            login,
+            userManagementAPIRequests,
+            exerciseAPIRequests,
+            page,
+            courseList,
+            courseOverview,
+            programmingExerciseOverview,
+        }) => {
+            await login(admin);
+            const response = await userManagementAPIRequests.getUser(studentFour.username);
+            const studentFourUser = await response.json();
+            await exerciseAPIRequests.createTeam(exercise.id!, [studentFourUser], tutorUser);
+
+            await login(studentFour, '/');
+            await page.waitForURL(/\/courses/);
+            await courseList.openCourse(course.id!);
+            await courseOverview.openExercise(exercise.title!);
+            await expect(programmingExerciseOverview.getCloneRepositoryButton()).not.toBeVisible();
+            await expect(programmingExerciseOverview.getExerciseDetails().getByText('Not yet started')).toBeVisible();
+            await courseOverview.startExercise(exercise.id!);
+            await expect(programmingExerciseOverview.getExerciseDetails().getByText('No graded result')).toBeVisible();
+        });
```

Comment:
This test relies on the order of the test execution to actually test what is described, as it assumes that the currently first test is executed first.
I think it would be better if all tests worked independently or if they were explicitly forced to be executed after each other.

I think to this test case it should be added that another team is created and makes a submission, as otherwise the test does not really test (independently from other tests) what the title describes

Reply:
Good point. I already forced them for serial execution, but isolating them makes sense, and actually they were all ready for that. So I just removed the serial part 😄